addons/l10n_it_edi/models/ir_mail_server.py

```diff
@@ -28,6 +28,19 @@ class FetchmailServer(models.Model):
     l10n_it_is_pec = fields.Boolean('PEC server', help="If PEC Server, only mail from '...@pec.fatturapa.it' will be processed.")
     l10n_it_last_uid = fields.Integer(string='Last message UID', default=1)
 
+    def _search_edi_invoice(self, att_name, send_state=False):
+        """ Search sent l10n_it_edi fatturaPA invoices """
+
+        conditions = [
+            ('edi_document_ids', '!=', False),
+            ('edi_document_ids.edi_format_id.code', '=', 'fattura_pa'),
+            ('edi_document_ids.attachment_id.name', '=', att_name),
+        ]
```

Comment:
This means take all the journal entries where there is at least one EDI document linked that has a code `'fattura_pa'` and where there is a least one EDI document that has an attachment with the name `att_name`.
Is that what was expected? Wasn't it to find EDI documents that have the format `'fattura_pa'` and the attachment name `att_name` at the same time?

Reply:
Hmmm... you are right on the conditions. Anyway, the code is looking for the move, not for the edi_document.
It will be hardly be the case that the same move has an attachment with a FatturaPA name as the receipt (filename has a pattern) and it's not for FatturaPA. 
The check before was just on "l10n_it_einvoice_name" so I guess the format ID can be a little redundant.

Would this be better?
```py
conditions = [('edi_format_id.code', '=', 'fattura_pa'), ('attachment_id.name', '=', att_name), ('move_id', "!=", False)]
self.env['account.edi.document'].search(conditions).move_id
```

Reply:
Looks better indeed